File: Data Structures & Algorithms/two-integer-sum/submission-2.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        std::map<int, int> map;
        for (int i = 0; i < nums.size(); i++) {
            if (map.find(nums[i]) != map.end()) {
                if (i != map[nums[i]]) {
                    std::vector<int> res;
                    res.push_back(map[nums[i]]);
                    res.push_back(i);
                    return res;
                }
            }
            map[target - nums[i]] = i;
        }
    }
};

File: Data Structures & Algorithms/minimum-size-subarray-sum/submission-2.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        l, currentSum = 0, 0
        minLength = float('infinity')

        for r in range(len(nums)):
            currentSum += nums[r]
            while currentSum >= target:
                minLength = min(minLength, r - l + 1)
                currentSum -= nums[l]
                l += 1

        return minLength if minLength != float('infinity') else 0